app/Http/Controllers/AdminController.php

```diff
@@ -26,4 +28,77 @@ public function users()
         $users = User::all(); // Retrieve all registered users
         return view('admin.users', compact('users'));
     }
+
+    public function updateUser(Request $request, $id)
+    {
+    $user = User::findOrFail($id);
+    // Update user details based on the request
+    $user->name = $request->input('name');
+    $user->middle_name = $request->input('middle_name');
+    $user->last_name = $request->input('last_name');
+    $user->gender = $request->input('gender');
+    $user->email = $request->input('email');
+    $user->user_name = $request->input('user_name');
+    $user->nickname = $request->input('nickname');
+    $user->date_of_birth = $request->input('date_of_birth');
+    $user->contact_number = $request->input('contact_number');
+    $user->zip_code = $request->input('zip_code');
+    $user->address = $request->input('address');
+
+    $user->save();
+
+    return redirect()->back()->with('success', 'User details updated successfully.');
+    }
+
+    public function exportUsers()
+    {
+    $users = User::all();
+
+    // Create a new CSV writer instance
+    $csv = Writer::createFromString('');
+
+    // Add a header row to the CSV
+    $csv->insertOne([
+        'Name', 'Middle Name', 'Last Name',  'Gender', 'Email', 'Username', 'Nickname', 'Date of Birth', 'Contact Number', 'Zip Code', 'Address',
+    ]);
+
+    // Add data rows to the CSV
+    foreach ($users as $user) {
+        $csv->insertOne([
+            $user->name,
+            $user->middle_name,
+            $user->last_name,
+            $user->gender,
+            $user->email,
+            $user->user_name,
+            $user->nickname,
+            $user->date_of_birth,
+            $user->contact_number,
+            $user->zip_code,
+            $user->address,
+        ]);
+    }
+
+    // Define the CSV file name
+    $filename = 'users.csv';
+
+    // Set appropriate headers for download
+    header('Content-Type: text/csv');
+    header("Content-Disposition: attachment; filename=\"$filename\"");
+
+    // Output the CSV content
+    echo $csv;
+    }
+
+    public function search(Request $request)
```

Comment:
- add comment method above
- make scope search in the User model
ex.
Models/User.php
public function scopeSearch()

usage:
User::search();